installer/cluster.go

```diff
@@ -46,6 +53,70 @@ func (c *BaseCluster) HandleAuthenticationFailure(cluster Cluster, err error) bo
 	return true
 }
 
+type receiveBackupWriter struct {
+	Callback func(read int)
+	read     int
+}
+
+func (w *receiveBackupWriter) Write(p []byte) (int, error) {
+	n := len(p)
+	w.read = w.read + n
+	w.Callback(w.read)
```

Comment:
I think this is slightly less work to use as a `Reader` instead of a `Writer`. Proxy the reads and do the tracking there.